server/controllers/bookingRequest.js

```diff
@@ -0,0 +1,121 @@
+const asyncHandler = require("express-async-handler");
+const User = require("../models/User");
+const BookingRequest = require("../models/BookingRequest");
+
+// @route POST /bookingrequests/create
+// @desc Create a new request
+// @access Private
+
+exports.createBookingRequest = asyncHandler(async (req, res, next) => {
+  try {
+    const userId = req.user.id;
+    const user = await User.findById(userId);
+    const { sitter_id, start, end, description } = req.body;
+    if (user) {
+      const sitter = await User.findById(sitter_id);
+      if (sitter) {
+        const bookingRequest = await BookingRequest.create({
+          user_id: user,
+          sitter_id: sitter,
+          start,
+          end,
+          description,
+        });
+
+        res.status(201).json({
+          success: {
+            bookingRequest: {
+              request_id: bookingRequest._id,
+              user_id: bookingRequest.user_id._id,
+              sitter_id: bookingRequest.sitter_id._id,
+              start: bookingRequest.start,
+              end: bookingRequest.end,
+              description: bookingRequest.description,
+            },
+          },
+        });
+      } else {
+        res.status(400);
+        throw new Error("Unable to find the dog sitter info");
+      }
+    } else {
+      res.status(400);
+      throw new Error("Unable to find the dog owner info");
+    }
+  } catch (error) {
+    res.status(500).json({
+      error,
+    });
+  }
+});
+
+// @route PUT /bookingrequests/update/:requestId
+// @desc Update a booking request (only for users who created a request)
+// @access Private
+
+exports.updateBookingRequest = asyncHandler(async (req, res, next) => {
+  try {
+    const userId = req.user.id;
+    const { requestId } = req.params;
+
+    if (requestId && requestId) {
+      const filter = { _id: requestId, user_id: userId };
+      const { ...requestUpdate } = req.body || {};
```

Comment:
Why spread this?

Reply:
resolved by 8b6f5ae3b1f1917fa51974715ad5de067694edcd